<a href="https://colab.research.google.com/github/drashtii27/DA_Python_Project/blob/main/Forbes_Americas_Top_Colleges.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Context

Starting in 2008, every year Forbes Magazine publishes a list of America's best colleges. When it comes to the question everyone seems to be asking, “Is college worth it? this published list of colleges comes handy to take a decision based on student's requirement or desire.

The mission of the college ranking by Forbes Magazine is to conduct an annual review of the undergraduate institutions that deliver the top academics, best experiences, career success and lowest debt. Whether a school is in the Top 10 or near the bottom of the list, the 650 colleges are the best in the country.

For most families, choosing a four-year college is one of the biggest and most expensive decisions they can make. For students, this time of their life may layout their future plans. So choose carefully.

## About Data

The data set contains the rankings of 650 Unites States colleges along with various other statistics pertaining to each school.

<img src ="https://i.ibb.co/sKwQjfw/Stanford.jpg" width=80% height =40%></img>


### Load required packages

In [51]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


### Load the data into pandas dataframe and view the details

In [2]:
df_college = pd.read_csv('ForbesAmericasTopColleges2019.csv')

### Dimension of the data

In [3]:
df_college.shape
#to know number of rows and columns

(650, 17)

In [4]:
print("The number of rows are: ",df_college.shape[0],"\nThe number of columns are: ",df_college.shape[1])

The number of rows are:  650 
The number of columns are:  17


### Datatype of the fields in the data

In [5]:
df_college.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 650 entries, 0 to 649
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Rank                      650 non-null    int64  
 1   Name                      650 non-null    object 
 2   City                      650 non-null    object 
 3   State                     650 non-null    object 
 4   Public/Private            650 non-null    object 
 5   Undergraduate Population  650 non-null    int64  
 6   Student Population        650 non-null    int64  
 7   Net Price                 648 non-null    float64
 8   Average Grant Aid         646 non-null    float64
 9   Total Annual Cost         650 non-null    int64  
 10  Alumni Salary             635 non-null    float64
 11  Acceptance Rate           648 non-null    float64
 12  SAT Lower                 551 non-null    float64
 13  SAT Upper                 551 non-null    float64
 14  ACT Lower 

Observations:

We have 5 fields with 'object' datatype, 8 fields of datatype 'float' and 4 fields with 'integer' datatype


### Check the missing values in the data
NA values, gets mapped to True values.Everything else gets mapped to False values.
'sum()' method followed by 'isnull()' adds up all the True instances per column to give a summary of missing values per column

In [6]:
df_college.isnull().sum()

Rank                         0
Name                         0
City                         0
State                        0
Public/Private               0
Undergraduate Population     0
Student Population           0
Net Price                    2
Average Grant Aid            4
Total Annual Cost            0
Alumni Salary               15
Acceptance Rate              2
SAT Lower                   99
SAT Upper                   99
ACT Lower                   97
ACT Upper                   97
Website                      5
dtype: int64

Observations:

We observe many missing values in SAT and ACT score fields. 

Let's drop 'SAT' and 'ACT' score columns for our analysis.

Also, there are missing values observed in 'Net Price', 'Average Grant Aid', 'Alumni Salary' and 'Acceptance Rate'. 

Let's drop the records/rows with missing values before proceeding with the analysis

'Website' is unique identifier of each college and is not required for further analysis. So, we can drop this column as well.



### Data Cleanup

Let's first drop all columns that are not need. 

'drop()' method with 'axis=1' indicates that columns have to be dropped. List of column names to be dropped is passed to the method. 'inplace = True' ensures that the columns are dropped from the DataFrame permanently

In [7]:
df_college.drop(['Website', 'SAT Lower','SAT Upper','ACT Lower','ACT Upper'], axis=1, inplace=True)

'dropna()' method drops all rows with NA values (by default axis = 0)

In [8]:
df_college.dropna(inplace=True)

In [9]:
df_college.shape

(628, 12)

Note:

After dropping the rows which had missing values we are left with 628 rows before dropping it was 650,17.

Also as we have dropped rows with missing values the index is no more continuous and has to be reset

### Reset the index of the dataframe

'reset_index()' resets the index of the DataFrame

'drop' as True, will not try to insert index into DataFrame columns

'inplace' as True resets the index permanently

In [10]:
df_college.reset_index(drop=True, inplace=True)

### Check the first 5 rows of the dataset


In [11]:
df_college.head()

Rank                                   Name       City State  \
0     1                     Harvard University  Cambridge    MA   
1     2                    Stanford University   Stanford    CA   
2     3                        Yale University  New Haven    CT   
3     4  Massachusetts Institute of Technology  Cambridge    MA   
4     5                   Princeton University  Princeton    NJ   

  Public/Private  Undergraduate Population  Student Population  Net Price  \
0        Private                     13844               31120    14327.0   
1        Private                      8402               17534    13261.0   
2        Private                      6483               12974    18627.0   
3        Private                      4680               11466    20771.0   
4        Private                      5659                8273     9327.0   

   Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  
0            49870.0              69600       146800.0              5.0  
1            50134.0              69109       145200.0              5.0  
2            50897.0              71290       138300.0              7.0  
3            43248.0              67430       155200.0              7.0  
4            48088.0              66150       139400.0              6.0

### Display the last five rows of the dataset


In [12]:
df_college.tail()

Rank                         Name         City State Public/Private  \
623   645            Morehouse College      Atlanta    GA        Private   
624   646  New Mexico State University   Las Cruces    NM         Public   
625   647     Indiana State University  Terre Haute    IN         Public   
626   648    Emory &amp; Henry College        Emory    VA        Private   
627   650              Catawba College    Salisbury    NC        Private   

     Undergraduate Population  Student Population  Net Price  \
623                      2202                2219    34928.0   
624                     13379               14432     8625.0   
625                     13626               13763    13012.0   
626                      1094                1226    19340.0   
627                      1336                1463    19807.0   

     Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  
623            17318.0              48723        98700.0             74.0  
624             9582.0              34720        96700.0             64.0  
625             9297.0              32938        85600.0             85.0  
626            27155.0              48100        70700.0             72.0  
627            23662.0              44921        78100.0             42.0

Observations:

From the head() and tail() output we now see that the indexes are realigned i.e. from 0 through 627 

### Recheck the dimension, info of the data and missing values of data

In [13]:
print("The number of rows are: ",df_college.shape[0],"\nThe number of columns are: ",df_college.shape[1])
df_college.info()

The number of rows are:  628 
The number of columns are:  12
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 628 entries, 0 to 627
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Rank                      628 non-null    int64  
 1   Name                      628 non-null    object 
 2   City                      628 non-null    object 
 3   State                     628 non-null    object 
 4   Public/Private            628 non-null    object 
 5   Undergraduate Population  628 non-null    int64  
 6   Student Population        628 non-null    int64  
 7   Net Price                 628 non-null    float64
 8   Average Grant Aid         628 non-null    float64
 9   Total Annual Cost         628 non-null    int64  
 10  Alumni Salary             628 non-null    float64
 11  Acceptance Rate           628 non-null    float64
dtypes: float64(4), int64(4), object(4)
memory usage: 59.0+ KB


In [14]:
df_college.isnull().sum()

Rank                        0
Name                        0
City                        0
State                       0
Public/Private              0
Undergraduate Population    0
Student Population          0
Net Price                   0
Average Grant Aid           0
Total Annual Cost           0
Alumni Salary               0
Acceptance Rate             0
dtype: int64

Note:

The data is now clean and has no missing values

# Data Analysis

### Get Summary of Data

'describe()' method generates descriptive statistics. Descriptive statistics include those that summarize the central
tendency, dispersion and shape of a dataset's distribution,unique values etc.

Passing 'include = 'all'' to describe() method provides descriptive summary of all fields in the DataFrame.


In [15]:
df_college.describe(include='all')

Rank                Name      City State Public/Private  \
count   628.000000                 628       628   628            628   
unique         NaN                 628       457    51              2   
top            NaN  Harvard University  New York    NY        Private   
freq           NaN                   1        14    64            384   
mean    322.122611                 NaN       NaN   NaN            NaN   
std     185.360460                 NaN       NaN   NaN            NaN   
min       1.000000                 NaN       NaN   NaN            NaN   
25%     161.750000                 NaN       NaN   NaN            NaN   
50%     321.500000                 NaN       NaN   NaN            NaN   
75%     479.250000                 NaN       NaN   NaN            NaN   
max     650.000000                 NaN       NaN   NaN            NaN   

        Undergraduate Population  Student Population     Net Price  \
count                 628.000000          628.000000    628.000000   
unique                       NaN                 NaN           NaN   
top                          NaN                 NaN           NaN   
freq                         NaN                 NaN           NaN   
mean                10235.472930        12312.912420  22481.570064   
std                 11228.048723        13246.660229   8099.771637   
min                   323.000000          406.000000   1298.000000   
25%                  2062.750000         2294.500000  16476.000000   
50%                  4825.500000         6538.500000  22065.000000   
75%                 15867.000000        18357.250000  27880.750000   
max                 65100.000000        75044.000000  47270.000000   

        Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  
count          628.000000         628.000000     628.000000       628.000000  
unique                NaN                NaN            NaN              NaN  
top                   NaN                NaN            NaN              NaN  
freq                  NaN                NaN            NaN              NaN  
mean         19971.522293       50722.579618   98616.242038        61.721338  
std          11214.740998       12670.126944   14026.078146        21.795857  
min           2975.000000        8646.000000   70700.000000         5.000000  
25%           9253.000000       40076.000000   88600.000000        49.000000  
50%          19605.000000       50478.000000   96300.000000        67.000000  
75%          27431.000000       60965.000000  105400.000000        78.000000  
max          50897.000000       75735.000000  158200.000000       100.000000

### 1) Which are the top 5 colleges listed in the Forbes Magazine 2019?

<img src ="https://i.ibb.co/dj0PFhK/Harvard.jpg" width=80% height =40%></img>


As we need the 'Name' of the colleges, we first pick the 'Name' column and then display the top 5 rows using head() function

We can also save the information in an object 'top5' and display it

In [16]:
top5 = df_college.Name.head() 
print(top5)

0                       Harvard University
1                      Stanford University
2                          Yale University
3    Massachusetts Institute of Technology
4                     Princeton University
Name: Name, dtype: object


Note:

Since the data is ordered as per the rank of the universities we can just pick up the top 5 colleges.

### 2) Which state and city do the top 5 colleges belong to?

As we need the 'Name', 'City' and 'State' of the colleges, we first pick the required columns and then display the top 5 rows using head() function

In [17]:
df_college[['Name','City','State']].head(5)

Name       City State
0                     Harvard University  Cambridge    MA
1                    Stanford University   Stanford    CA
2                        Yale University  New Haven    CT
3  Massachusetts Institute of Technology  Cambridge    MA
4                   Princeton University  Princeton    NJ

will try to extract rows with the help of loc and iloc

In [18]:
df_college.iloc[0:5,0:4]

Rank                                   Name       City State
0     1                     Harvard University  Cambridge    MA
1     2                    Stanford University   Stanford    CA
2     3                        Yale University  New Haven    CT
3     4  Massachusetts Institute of Technology  Cambridge    MA
4     5                   Princeton University  Princeton    NJ

Note:

As 'City' and 'State' are immediate columns after 'Name', iloc can also be used to get the results

### 3) How many states and which all states have the colleges listed by Forbes?

'nunique()' returns number of unique elements in the object. As we want number of unique states, we first pick up the 'State' column of the DataFrame and then call nunique() function

In [19]:
df_college.State.nunique()

51

'unique()' function returns unique values of Series object. Hence, we first pick up the 'State' column of the DataFrame and then call unique() function to get all all the States

In [20]:
df_college.State.unique()

array(['MA', 'CA', 'CT', 'NJ', 'PA', 'RI', 'NC', 'NH', 'NY', 'DC', 'IL',
       'IN', 'MI', 'TX', 'MD', 'ME', 'TN', 'MO', 'VA', 'VT', 'MN', 'GA',
       'WA', 'WI', 'FL', 'OH', 'IA', 'CO', 'UT', 'OR', 'LA', 'SC', 'DE',
       'AL', 'AZ', 'OK', 'NE', 'KY', 'KS', 'AR', 'WY', 'NV', 'MS', 'ND',
       'HI', 'MT', 'SD', 'NM', 'ID', 'WV', 'AK'], dtype=object)

### 4) In all how many private and public colleges are there in the given list?

'Public/Private' column is of object data type in the DataFrame.

'value_count()' returns a series containing counts of unique values

In [21]:
df_college['Public/Private'].value_counts()

Private    384
Public     244
Name: Public/Private, dtype: int64

Hence, there are 384 private and 244 public colleges in the given list

### 5) From the list how many colleges are there within each State? Which State has highest number of colleges?

As we have to find number of colleges within each State, we first group (groupby()) the data based on 'State'. For each State we then count the colleges using number of unique values (nunique())  in 'Name' field.

As we want the 'State' with highest number of colleges, we sort the values (sort_values()) in descending order to get the results.

<img src ="https://i.ibb.co/GcnXVC0/New-York.jpg" width=80% height =40%></img>

In [22]:
top_states=df_college.groupby(by='State').Name.nunique().sort_values(ascending = False).head()
print(top_states)

State
NY    64
CA    56
PA    39
MA    31
OH    29
Name: Name, dtype: int64


### 6) What is percentage of undergraduates in each college? (Add a new field to calculate)

Divsion of 'Undergraduate Population' by 'Student Population' multiplied by 100 will give percentage of undergraduates.

round() method will round off the derived value to 2 decimal places which is then stored in new field 'Percentage UG'

In [23]:
df_college['Percentage UG'] = round(df_college['Undergraduate Population']/df_college['Student Population']*100,2)

In [24]:
df_college.head()

Rank                                   Name       City State  \
0     1                     Harvard University  Cambridge    MA   
1     2                    Stanford University   Stanford    CA   
2     3                        Yale University  New Haven    CT   
3     4  Massachusetts Institute of Technology  Cambridge    MA   
4     5                   Princeton University  Princeton    NJ   

  Public/Private  Undergraduate Population  Student Population  Net Price  \
0        Private                     13844               31120    14327.0   
1        Private                      8402               17534    13261.0   
2        Private                      6483               12974    18627.0   
3        Private                      4680               11466    20771.0   
4        Private                      5659                8273     9327.0   

   Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  \
0            49870.0              69600       146800.0              5.0   
1            50134.0              69109       145200.0              5.0   
2            50897.0              71290       138300.0              7.0   
3            43248.0              67430       155200.0              7.0   
4            48088.0              66150       139400.0              6.0   

   Percentage UG  
0          44.49  
1          47.92  
2          49.97  
3          40.82  
4          68.40

In [25]:
df_college[['Name','Percentage UG']].head()

Name  Percentage UG
0                     Harvard University          44.49
1                    Stanford University          47.92
2                        Yale University          49.97
3  Massachusetts Institute of Technology          40.82
4                   Princeton University          68.40

Note:

New field gets added at end of the DataFrame

### 7) Which are the 5 colleges with highest percentage of undergraduates?

Let's just pick up 'Name' and 'Percentage UG' field, sort by 'Percentage UG' in descending order to get the desired ouput

In [26]:
df_college[['Name','Percentage UG']].sort_values(by='Percentage UG', ascending = False).head()

Name  Percentage UG
556  University of Texas, Rio Grande Valley          99.99
246            Saint John's University (MN)          99.94
552                      Oakland University          99.94
357                    Salisbury University          99.92
338                     St. Norbert College          99.86

### 8) 5 colleges that receive highest Average Grant Aid?

Sort 'Average Grant Aid' in descending order and pick up the top 5 colleges using 'Name' field

<img src ="https://i.ibb.co/TMtRJ54/Yale.jpg" width=80% height =40%></img>

In [27]:
df_college.sort_values(by='Average Grant Aid', ascending = False).Name.head(5)

2         Yale University
1     Stanford University
0      Harvard University
26        Amherst College
13    Columbia University
Name: Name, dtype: object


### 9) What is the average Undergraduate population in State of 'CA'?

First lets filter the DataFrame and get only rows with 'State' as 'CA'

We then pick up the 'Undergraduate Population' for the above subset and calculate its mean

round off the final value to 2 decimal places using the round() function

In [28]:
round(df_college[df_college.State == 'CA']['Undergraduate Population'].mean(),2)

13186.91

### 10) Which are the colleges where chances of admission is high?

As per the data dictionary 'Acceptance Rate' field provides information on chances of admission. We can sort this column in descending order to get the results

In [29]:
df_college.sort_values(by='Acceptance Rate', ascending = False).head(5)

Rank                                Name         City State  \
588   606        University of Texas, El Paso      El Paso    TX   
607   627  University of Southern Mississippi  Hattiesburg    MS   
339   347               University of Wyoming      Laramie    WY   
616   638             Wright State University       Dayton    OH   
300   308              University of Kentucky    Lexington    KY   

    Public/Private  Undergraduate Population  Student Population  Net Price  \
588         Public                     24386               25078     8135.0   
607         Public                     13606               14478    13178.0   
339         Public                     10874               12397    13070.0   
616         Public                     14332               15957    16249.0   
300         Public                     23718               29465    18527.0   

     Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  \
588             9183.0              36955        92100.0            100.0   
607             9292.0              24883        83900.0             98.0   
339             7422.0              31387        98800.0             97.0   
616             6546.0              32506        87800.0             97.0   
300             9387.0              45634        96400.0             96.0   

     Percentage UG  
588          97.24  
607          93.98  
339          87.71  
616          89.82  
300          80.50

### 11) Find if there are any 'Art' colleges in the list

To find a specific string value 'Art' we will use the 'contains()' method. We will pass the string in the paranthesis which needs to be located in the dataframe variable.

In [30]:
df_college[df_college['Name'].str.contains("Art")]

Rank                               Name           City State  \
295   303  Maryland Institute College of Art      Baltimore    MD   
393   402   California Institute of the Arts  Santa Clarita    CA   

    Public/Private  Undergraduate Population  Student Population  Net Price  \
295        Private                      2041                2128    38763.0   
393        Private                       991                1498    46101.0   

     Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  \
295            18939.0              62850        84500.0             62.0   
393            15401.0              68993        91000.0             24.0   

     Percentage UG  
295          95.91  
393          66.15

### 12) Lets divide the Alumni Salary equally into 3 buckets (Low Salary - L; Medium Salary - M; High Salary - H ) and check which colleges falls in 'H' bucket

In [31]:
df_college_new = df_college.copy()

Take a copy of the original DataFrame to work on

In [89]:
df_college_new['Alumni Sal Level']=pd.cut(df_college_new['Alumni Salary'], 3, labels =["L", "M", "H"])


'cut()' function bins values into discrete intervals. 'cut()' function is used when you need to segment and sort data values into bins. This function is also useful for going from a continuous variable to a categorical variable.

In the above code we have created 3 bins using 'Alumni Salary' column. Lowest range of salary is labelled as 'L', middle range is labelled as 'M' and highest range of salary is labelled as 'H'. These labels are saved in a new column 'Alumni Sal Level'.

Note: The entire range of the 'Alumni Salary' has been divided into 3 equal ranges i.e. 3 bins

In [33]:
df_college_new.head()


Rank                                   Name       City State  \
0     1                     Harvard University  Cambridge    MA   
1     2                    Stanford University   Stanford    CA   
2     3                        Yale University  New Haven    CT   
3     4  Massachusetts Institute of Technology  Cambridge    MA   
4     5                   Princeton University  Princeton    NJ   

  Public/Private  Undergraduate Population  Student Population  Net Price  \
0        Private                     13844               31120    14327.0   
1        Private                      8402               17534    13261.0   
2        Private                      6483               12974    18627.0   
3        Private                      4680               11466    20771.0   
4        Private                      5659                8273     9327.0   

   Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  \
0            49870.0              69600       146800.0              5.0   
1            50134.0              69109       145200.0              5.0   
2            50897.0              71290       138300.0              7.0   
3            43248.0              67430       155200.0              7.0   
4            48088.0              66150       139400.0              6.0   

   Percentage UG Alumni Sal Level  
0          44.49                H  
1          47.92                H  
2          49.97                H  
3          40.82                H  
4          68.40                H

In [34]:
df_college_new['Alumni Sal Level'].value_counts()

L    383
M    217
H     28
Name: Alumni Sal Level, dtype: int64

Note:
There are only 28 colleges in the High Alumni Salary Range

### 13) Let's check the mean Alumni Salary at each level

In [35]:
round(df_college_new.groupby('Alumni Sal Level')['Alumni Salary'].mean(),2)

Alumni Sal Level
L     89835.77
M    109086.64
H    137575.00
Name: Alumni Salary, dtype: float64

### 14) Let's create a new dataframe to further analyze the 28 colleges with High Alumni Salary 

In [36]:
df_high_alumnisal = df_college_new[df_college_new['Alumni Sal Level'] == 'H']

'T' property transposes the table which helps in reviewing the statistics per column

In [37]:
df_high_alumnisal.describe(include='all').T

count unique                 top freq          mean  \
Rank                      28.0    NaN                 NaN  NaN     68.857143   
Name                        28     28  Harvard University    1           NaN   
City                        28     27           Cambridge    2           NaN   
State                       28     12                  CA    6           NaN   
Public/Private              28      2             Private   22           NaN   
Undergraduate Population  28.0    NaN                 NaN  NaN   6389.535714   
Student Population        28.0    NaN                 NaN  NaN  10603.321429   
Net Price                 28.0    NaN                 NaN  NaN  25288.571429   
Average Grant Aid         28.0    NaN                 NaN  NaN  32340.535714   
Total Annual Cost         28.0    NaN                 NaN  NaN  62952.464286   
Alumni Salary             28.0    NaN                 NaN  NaN      137575.0   
Acceptance Rate           28.0    NaN                 NaN  NaN     28.035714   
Percentage UG             28.0    NaN                 NaN  NaN     68.703929   
Alumni Sal Level            28      1                   H   28           NaN   

                                   std       min       25%       50%  \
Rank                         87.500144       1.0      8.75      41.5   
Name                               NaN       NaN       NaN       NaN   
City                               NaN       NaN       NaN       NaN   
State                              NaN       NaN       NaN       NaN   
Public/Private                     NaN       NaN       NaN       NaN   
Undergraduate Population   6290.912467     847.0    2759.0    4752.0   
Student Population        10225.926422     861.0    3253.0    7019.5   
Net Price                  9526.081595    6758.0  17913.75   24977.5   
Average Grant Aid         14681.882361    3597.0   20604.0   36200.5   
Total Annual Cost         14203.545967    8646.0   65024.0   67931.0   
Alumni Salary              7667.566372  129500.0  132025.0  135100.0   
Acceptance Rate              21.489846       5.0      9.75      22.0   
Percentage UG                  19.0947     40.82     50.03    69.785   
Alumni Sal Level                   NaN       NaN       NaN       NaN   

                               75%       max  
Rank                         93.25     333.0  
Name                           NaN       NaN  
City                           NaN       NaN  
State                          NaN       NaN  
Public/Private                 NaN       NaN  
Undergraduate Population   7001.75   31133.0  
Student Population        13197.75   41891.0  
Net Price                  34275.0   40376.0  
Average Grant Aid         44877.75   50897.0  
Total Annual Cost          69918.5   74428.0  
Alumni Salary             140100.0  158200.0  
Acceptance Rate               45.0      69.0  
Percentage UG              80.0725      99.6  
Alumni Sal Level               NaN       NaN

Observations:

Out of the 28 shortlisted colleges 22 are private colleges

'CA' is the State which has higest number of these colleges

'Net Price' of these colleges ranges from minimum of 6758 to maximum of 40376

Average 'Acceptance Rate' of these colleges is 28.04


**Find** **which colleges are affordable
.**

---



In [50]:
df_af=df_college.sort_values(by='Net Price',ascending = False).head()
df_af.set_index('Rank')
df_af.sort_index()

Rank                              Name           City State  \
107   112     Rhode Island School of Design     Providence    RI   
215   222                   Emerson College         Boston    MA   
320   328                   Pratt Institute       Brooklyn    NY   
393   402  California Institute of the Arts  Santa Clarita    CA   
395   404          Berklee College of Music         Boston    MA   

    Public/Private  Undergraduate Population  Student Population  Net Price  \
107        Private                      2030                2440    46277.0   
215        Private                      4088                4459    43965.0   
320        Private                      4050                5035    43969.0   
393        Private                       991                1498    46101.0   
395        Private                      6762                6952    47270.0   

     Average Grant Aid  Total Annual Cost  Alumni Salary  Acceptance Rate  \
107            26169.0              67620       100600.0             32.0   
215            20323.0              65324        94300.0             46.0   
320            20362.0              65308       102300.0             57.0   
393            15401.0              68993        91000.0             24.0   
395            19811.0              63574        84200.0             53.0   

     Percentage UG  
107          83.20  
215          91.68  
320          80.44  
393          66.15  
395          97.27

Observations:

Rhode Island School Of design is most affordable also its rank is 112 which is good enough for a design college

It avg net price is 46277 and also provide grant aid of 26169 .

So for students who are finding affordable college in design Rhode Island School Of design should be recommended.

